app/controllers/request_controller.rb

```diff
@@ -862,28 +878,27 @@ def check_batch_requests_and_user_allowed
   end
 
   def render_new_compose(batch)
-
     params[:info_request] = { } if !params[:info_request]
 
-    # Read parameters in
+    # Reconstruct the params
     unless batch
       # first the public body (by URL name or id)
-      if params[:url_name]
-        if params[:url_name].match(/^[0-9]+$/)
-          params[:info_request][:public_body] = PublicBody.find(params[:url_name])
-        else
-          public_body = PublicBody.find_by_url_name_with_historic(params[:url_name])
-          raise ActiveRecord::RecordNotFound.new("None found") if public_body.nil? # TODO: proper 404
-          params[:info_request][:public_body] = public_body
+      params[:info_request][:public_body_id] ||=
+        if params[:url_name]
+          if params[:url_name].match(/^[0-9]+$/)
```

Comment:
Haven't really modified this line, so don't want to do anything that the original didn't do.